ludwig/hyperopt/execution.py

```diff
@@ -819,15 +844,92 @@ def get_build_hyperopt_executor(executor_type):
 executor_registry = {"ray": RayTuneExecutor}
 
 
-def set_values(model_dict, name, parameters_dict):
-    if name in parameters_dict:
-        params = parameters_dict[name]
-        for key, value in params.items():
-            if isinstance(value, dict):
-                for sub_key, sub_value in value.items():
-                    model_dict[key][sub_key] = sub_value
-            else:
-                model_dict[key] = value
+def set_values(params: Dict[str, Any], model_dict: Dict[str, Any]):
+    for key, value in params.items():
+        if isinstance(value, dict):
+            for sub_key, sub_value in value.items():
+                if key not in model_dict:
+                    model_dict[key] = dict()
+                model_dict[key][sub_key] = sub_value
+        else:
+            model_dict[key] = value
+
+
+def update_model_dict_with_shared_params(
+    model_dict: Dict[str, Any],
+    trial_parameters_dict: Dict[str, Dict[str, Any]],
+    config_feature_group: str = None,
+    shared_params_features_dict: Dict[str, Dict[str, Set]] = None,
+):
+    """Updates the parameters of feature_name in model_dict based on hyperopt parameters sampled.
+
+    :param model_dict: Underlying config for the specific input/output feature populated with defaults and feature
+            specific parameters. This may be updated with values from the hyperopt search space.
```

Comment:
```suggestion
    :param model_dict: Underlying config for the specific input/output feature populated with potentially a mix of default and feature-specific parameters. This may be updated with values from the hyperopt search space.
```